test/operators/test_fermionic_op.py

```diff
@@ -18,73 +18,93 @@
 
 from ddt import data, ddt, unpack
 
-from qiskit_nature import QiskitNatureError
 from qiskit_nature.operators import FermionicOp
 
 
 def fermion_labels(length):
     """Generate list of fermion labels with given length."""
-    return [
-        "".join(label) for label in product(["I", "+", "-", "N", "E"], repeat=length)
-    ]
+    return ["".join(label) for label in product(["I", "+", "-", "N", "E"], repeat=length)]
 
 
 @ddt
 class TestFermionicOp(QiskitNatureTestCase):
     """FermionicOp tests."""
 
+    def assertFermionEqual(self, first: FermionicOp, second: FermionicOp):
```

Comment:
I think it may be useful to implement `__eq__` in the `FermionicOp` class instead of this.

Reply:
This is not true equality. Users will be confused.